editor/block-settings-menu/content.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { IconButton } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { isEditorSidebarOpened } from '../selectors';
+import { selectBlock } from '../actions';
+
+function BlockSettingsMenuContent( { onDelete, onSelect, isSidebarOpened, toggleSidebar, setActivePanel } ) {
+	const toggleInspector = () => {
```

Comment:
As much as I dislike the class component, I feel it has the distinct advantage in avoiding reconciliation which will occur here on assigning the new function reference for every render. Maybe the supposed performance benefits of function components counter-act my own concerns though.

Reply:
Yes, I tend to avoid the class components as much as I can, I personally don't mind these "rerenderings" but I don't have any proof saying the functional components even with those functions are faster.

Should we try to set a guideline, or is it fine to have separate styles here?

Reply:
Rumors say that functional component was slower in React 15, but it should be comparable in React 16 on the initial render.

There is only one prop which can trigger re-render of this component `isSidebarOpened` which will cause the first icon to re-render because of this dynamically created function. There is also another option here, we could create this function as 3rd param in the `connect` function.

I commented this already in other PR. I don't like the fact that we have a choice of the component syntax in the first. It seems like class component is a better choice here, but it's hard to measure performance implications. We could use `mergeProps` from `connect` more often, but this brings another set of issues here :)

Reply:
> Rumors say that functional component was slower in React 15, but it should be comparable in React 16, on the initial render.

Actually, I saw a benchmark showing that the functional components are 15% faster than the class components but unless you benchmark your self, this is irrelevant.

I'm very much in favor of avoiding premature optimization vs simpler code. And I have to admit I don't like classes that much.